nemo/collections/nlp/models/dialogue/dialogue_gpt_classification_model.py

```diff
@@ -294,7 +362,16 @@ def forward(self, input_ids, attention_mask, labels, inference=True):
             labels_mask = labels_mask_0 > 0
 
             loss = self.mask_and_reduce_loss(labels_mask, unmasked_unreduced_loss)
-        return loss
+            loss_per_sample = self.mask_and_reduce_loss_per_sample(labels_mask, unmasked_unreduced_loss)
+
+        return loss, loss_per_sample
+
+    def mask_and_reduce_loss_per_sample(self, loss_mask, output_tensor):
```

Comment:
How is this different from `mask_and_reduce_loss`?

Reply:
It is for `Mask and reduce loss based on each sample in batch. Useful for ranking candidates with the same prompt in batch based on loss` Also address as a comment in code